### PR TITLE
Split CI in different steps and jobs

### PR DESCRIPTION
**Pull request by jaimergp**
- [X] Tag issue being addressed
- [ ] Add [tests](https://github.com/openforcefield/openforcefield/tree/master/openforcefield/tests)
- [ ] Update docstrings/[documentation](https://github.com/openforcefield/openforcefield/tree/master/docs), if applicable
- [ ] Update [changelog](https://github.com/openforcefield/openforcefield/blob/master/docs/releasehistory.rst)

Addresses #561 

Proposed split is:
* Matrix job (runs on several configs):
    * Unit tests + Examples (two separate steps for easy diagnosis)
* Single config job (Ubuntu, Py37, with rdkit and openeye): 
    * Lint checks (new! run `flake8`, `isort --check` and `black --check`, just to see how it feels)
    * Documentation (test readme links and build sphinx locally)

## Questions

- [ ] Do we want Black checks? If so, line length? Maybe the default 88?
- [ ] Do we want Flake8 checks? It's catching a bunch of (important, I'd say) stuff. We'd need to agree on a good config.
- [ ] Do we want to turn Sphinx warnings (there are a lot!) into errors?

### CODE REVIEW

**Comment by mattwthompson** [comment]
>  Do we want Black checks? If so, line length? Maybe the default 88?
 Do we want Flake8 checks? It's catching a bunch of (important, I'd say) stuff. We'd need to agree on a good config.

I would personally like at least `flake8` to be implemented eventually and I agree that we need agree on the details before linting the existing code or enforcing it as a check. For example, I like prefer `black -S` but I may be alone in that. For now, can linters just operate on the diff?

Can we reach a consensus during the developer week?

**Comment by SimonBoothroyd** [comment]
> For example, I like prefer black -S but I may be alone in that. For now, can linters just operate on the diff?

My +1 is to just stick directly to the black defaults and not make any config changes. "Black is opinionated so you don’t have to be."

File: openforcefield/tests/conftest.py [hunk 1]
```diff
@@ -19,6 +19,15 @@
...
 import pytest
 
+def pytest_configure(config):
+    """
+    Initialization hook to register custom markers without a pytest.ini
+    More info: https://docs.pytest.org/en/latest/reference.html#initialization-hooks
+    """
+    config.addinivalue_line(
+        "markers", "slow: marks tests as slow (deselect with `-m 'not slow'`)"
+    )
```
Comment (j-wags) [inline openforcefield/tests/conftest.py:29]: 🦜

Comment (j-wags) [reply openforcefield/tests/conftest.py:29]: (To clarify -- The parrot is a very good thing)

**Comment by mattwthompson** [comment]
Okay, I've scaled this back a little just to get it over the finish line (linting will still happen, just splititng this off into smaller PRs)

* Uses miniconda action
* Install RDKit vs OpenEye array using separate files, not lines in script
* Includes `openeye-toolkits=2019` pin
* Split off tests into a few smaller steps

It's tempting to further split off the unit tests (one for each of the main modules) but I can load up the unit test run without my browser freezing up, so consider that a win.

**Comment by j-wags** [comment]
I realized that this is a lot of really good stuff, but I have just been flying from fire to fire the last few weeks and haven't reviewed this or had bandwidth to discuss the questions above. These changes will make everyone's lives a lot easier, so I don't want to block this any further.

I'm removing myself as reviewer. @mattwthompson can approve+merge this at his discretion! Thanks everyone for the hard work here.

**Pull request merged**